File: surface/widgets/nativescript/components/widgets/navigation/Pagination.ts
// ============================================================
// Clef Surface NativeScript Widget — Pagination
//
// Page navigation with previous/next and page numbers.
// ============================================================

import { StackLayout, Label, Button } from '@nativescript/core';

export interface PaginationProps {
  currentPage?: number;
  totalPages: number;
  siblingCount?: number;
  disabled?: boolean;
  onPageChange?: (page: number) => void;
  size?: 'sm' | 'md' | 'lg';
}

export function createPagination(props: PaginationProps): StackLayout {
  const { currentPage = 1, totalPages, siblingCount = 1, disabled = false, onPageChange, size = 'md' } = props;
  const container = new StackLayout();
  container.className = `clef-widget-pagination clef-size-${size}`;
  container.orientation = 'horizontal';
  container.accessibilityRole = 'navigation';
  container.accessibilityLabel = 'Pagination';

  const prevBtn = new Button();
  prevBtn.text = '\u2190';
  prevBtn.isEnabled = !disabled && currentPage > 1;
  prevBtn.accessibilityLabel = 'Previous page';
  prevBtn.on('tap', () => onPageChange?.(currentPage - 1));
  container.addChild(prevBtn);

  for (let i = 1; i <= totalPages; i++) {
    const pageBtn = new Button();
    pageBtn.text = String(i);
    pageBtn.className = i === currentPage ? 'clef-page-active' : 'clef-page';
    pageBtn.isEnabled = !disabled;
    pageBtn.accessibilityLabel = `Page ${i}`;
    pageBtn.on('tap', () => onPageChange?.(i));
    container.addChild(pageBtn);
  }

  const nextBtn = new Button();
  nextBtn.text = '\u2192';
  nextBtn.isEnabled = !disabled && currentPage < totalPages;
  nextBtn.accessibilityLabel = 'Next page';
  nextBtn.on('tap', () => onPageChange?.(currentPage + 1));
  container.addChild(nextBtn);

  return container;
}

export default createPagination;

File: surface/widgets/swiftui/components/widgets/navigation/Pagination.swift
// ============================================================
// Clef Surface SwiftUI Widget — Pagination
//
// Page navigation control displaying page numbers with
// previous/next buttons and optional ellipsis for large ranges.
// ============================================================

import SwiftUI

// --------------- Component ---------------

/// Pagination view for page navigation control.
///
/// - Parameters:
///   - currentPage: Binding to the current page number (1-based).
///   - totalPages: Total number of pages.
///   - siblingCount: Number of sibling pages to show around current.
///   - onPageChange: Callback when the page changes.
struct PaginationView: View {
    @Binding var currentPage: Int
    var totalPages: Int
    var siblingCount: Int = 1
    var onPageChange: ((Int) -> Void)? = nil

    private var pages: [PageItem] {
        var result: [PageItem] = []
        let start = Swift.max(1, currentPage - siblingCount)
        let end = Swift.min(totalPages, currentPage + siblingCount)

        if start > 1 {
            result.append(.page(1))
            if start > 2 { result.append(.ellipsis) }
        }

        for i in start...end {
            result.append(.page(i))
        }

        if end < totalPages {
            if end < totalPages - 1 { result.append(.ellipsis) }
            result.append(.page(totalPages))
        }

        return result
    }

    var body: some View {
        HStack(spacing: 4) {
            // Previous button
            SwiftUI.Button(action: {
                let prev = currentPage - 1
                currentPage = prev
                onPageChange?(prev)
            }) {
                Image(systemName: "chevron.left")
            }
            .disabled(currentPage <= 1)

            // Page buttons
            ForEach(Array(pages.enumerated()), id: \.offset) { _, item in
                switch item {
                case .page(let num):
                    SwiftUI.Button(action: {
                        currentPage = num
                        onPageChange?(num)
                    }) {
                        Text("\(num)")
                            .font(.body)
                            .fontWeight(num == currentPage ? .bold : .regular)
                            .foregroundColor(num == currentPage ? .white : .primary)
                            .frame(minWidth: 36, minHeight: 36)
                            .background(num == currentPage ? Color.accentColor : Color.clear)
                            .clipShape(RoundedRectangle(cornerRadius: 6))
                    }
                    .buttonStyle(.plain)

                case .ellipsis:
                    Text("...")
                        .foregroundColor(.secondary)
                        .frame(minWidth: 36, minHeight: 36)
                }
            }

            // Next button
            SwiftUI.Button(action: {
                let next = currentPage + 1
                currentPage = next
                onPageChange?(next)
            }) {
                Image(systemName: "chevron.right")
            }
            .disabled(currentPage >= totalPages)
        }
    }

    private enum PageItem {
        case page(Int)
        case ellipsis
    }
}
